PantheonCMD/pcbuild.py

```diff
@@ -82,28 +98,52 @@ def prepare_build_directory():
     os.makedirs('build/images')
 
 
-def coalesce_document(main_file, attributes=None):
+def resolve_attribute_tree(content, attributes):
+    """Attempt to recursively resolve AsciiDoc attributes into a string."""
+
+    continue_processing = True
+
+    while continue_processing:
+        if re.match(r'.*\{\S+\}.*', content):
+            for attribute in re.search(r'\{(.*?)\}', content).groups():
+                if attribute in attributes.keys():
+                    content = content.replace('{' + attribute + '}',attributes[attribute])
+                else:
+                    continue_processing = False
+        else:
+            continue_processing = False
+    return content
+
+
+def coalesce_document(main_file, attributes=None, depth=1, top_level=True):
     """Combines the content from includes into a single, contiguous source file."""
     attributes = attributes or {}
     lines = []
 
     # Open the file, iterate over lines
     if os.path.exists(main_file):
         with open(main_file) as input_file:
+            # Iterate over content
             for line in input_file:
                 # Process comments
                 if line.startswith('//'):
                     continue
+                # Process section depth
+                if line.strip().startswith('= '):
```

Comment:
This will catch level 2 headings (`== Some heading`) and count it as 1 level of depth, while it should be 2 levels

Reply:
Great catch @Levi-Leah - I have now updated 'coalesce_document' so that the default depth is '0', and it parses out options in a more detailed way. Rather than replacing the '=' symbols entirely, it now takes whatever is there and adds on X number of '=' symbols for the specified depth, if any. This way, if users specify the depth in the file itself, that value will be respected, and if they specify a leveloffset, that value will also get applied.